tests/shell/tests.py

```diff
@@ -29,3 +39,17 @@ def test_stdin_read(self, select):
             stdin.seek(0)
             call_command('shell')
         self.assertEqual(stdout.getvalue().strip(), '100')
+
+    @unittest.skipIf(IPython is not None, 'IPython is installed')
+    @mock.patch('django.core.management.commands.shell.select.select')
```

Comment:
Have added a comment.  Putting the patching in `setUp()` won't work since we don't want to patch for `test_stdin_read`.